### ISSUE TITLE
Width of input can't be set within a vstack.

### ISSUE THREAD

**Issue by hjpr** [reflex-dev/reflex#2861]
Setting width of a high level input component from within an vstack or other non radix container will not change width of input component. Changing the container to a flex allows input width to be 100% or set manually.

**Specifics (please complete the following information):**
 - Python Version: 3.11
 - Reflex Version: 0.4.4
 - OS: Ubuntu 23.10

**Comment by wassafshahzad** [comment]
Could you share a reproducible example ?

**Comment by hjpr** [comment]
def example() -> rx.Component:
    return rx.vstack(
        rx.input(
            placeholder='Test Example',
            width='100%'
        )
    )

Above is an approximate of how I discovered it. If you wrap in flex instead, the width is set appropriately. The rendered code appears to not carry width as a CSS prop assigned to input using vstack, but does carry using flex container.

**Comment by wassafshahzad** [comment]
I looked into the issue using the following code 
```
def index() -> rx.Component:
    return rx.vstack(
        rx.input(
            placeholder='Test Example',
            width='100%'
        )
    )
```
We get 
![vstacck_input](https://github.com/reflex-dev/reflex/assets/25094157/693156af-a336-4bb9-b192-e3b25b8c999c)

While using the following code 
```
def index() -> rx.Component:
    return rx.flex(
        rx.input(
            placeholder='Test Example',
            width='100%'
        )
    )

```
We get 
![flex_inut](https://github.com/reflex-dev/reflex/assets/25094157/f74ac07c-e248-4930-b1f2-a8ad1be8b029)

I both cases the input tag does have the width property although The TextFieldRoot does not and it seems to be consistent in both flex and vstack.

**Comment by spac3-monk3y** [comment]
Any workaround this limitation?

**Comment by masenf** [comment]
The workaround is to set the width in units or to set the 100% width on the `rx.input.root`

```python
rx.input.root(
    rx.input(),
    width="100%",
)
```

I hear this is one of things that is fixed in radix 3.0.0 which we'll be upgrading to soon.

**Comment by picklelo** [comment]
We've upgraded our input component, so this should work now with needing `rx.input.root`.
